code-scanning/roxctl.yml

```diff
@@ -0,0 +1,57 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# This workflow integrates roxctl with GitHub's Code Scanning
+# feature.
+#
+# roxctl is the CLI of StackRox, a container security platform.
+# roxctl will scan container images for vulnerabilities.
+#
+# See https://www.stackrox.io/docs/ for more on StackRox.
+#
+# See https://github.com/stackrox/roxctl-installer-action for
+# more on the roxctl-installer action and its parameters.
+
+name: Roxctl
+
+on:
+  push:
+    branches: [ $default-branch, $protected-branches ]
+  pull_request:
+    branches: [ $default-branch ]
+  schedule:
+    - cron: $cron-weekly
+
+permissions:
+  contents: read
+
+jobs:
+  roxctl-image-scan:
+    permissions:
+      # Only required for private repositories by github/codeql-action/upload-sarif
+      # to get the Action run status.
+      actions: read
+      # Required for actions/checkout to fetch code.
+      contents: read
+      # Require for github/codeql-action/upload-sarif to upload SARIF results.
+      security-events: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v4
+
+      - name: Setup roxctl
```

Comment:
Should we include `central-login` action here? Otherwise I think we should add an explicit `ROX_API_TOKEN` env var below.

Reply:
Yeah I was contemplating skipping this and maybe just adding some markers for the API token and such, but I added it now. Not sure whether the upstream contrib will be accepted with two actions being used, but I suppose we'll see (it's a pity that we need to connect to Central _anyways_ especially if you look at how easy it is to use the other tools).